hack/deploy-logging.sh

```diff
@@ -229,109 +307,90 @@ fi
-            -e '/quay.io\/openshift\/origin-logging-/ {s,quay.io/openshift/origin-,'"$registry"',}' | \
-        oc set env -e - deploy/cluster-logging-operator
-        # special handling for rsyslog for now
-        oc set env deploy/cluster-logging-operator RSYSLOG_IMAGE=${registry}logging-rsyslog:latest
-    # update the images to use in the CLO
-    else
-        oc set env deploy/cluster-logging-operator \
-            ELASTICSEARCH_IMAGE=$( construct_image_name logging-elasticsearch5 latest ) \
-            KIBANA_IMAGE=$( construct_image_name logging-kibana5 latest ) \
-            CURATOR_IMAGE=$( construct_image_name logging-curator5 latest ) \
-            FLUENTD_IMAGE=$( construct_image_name logging-fluentd latest ) \
-            RSYSLOG_IMAGE=$( construct_image_name logging-rsyslog latest )
-        cloimg=$( construct_image_name cluster-logging-operator latest )
-        eoimg=$( construct_image_name elasticsearch-operator latest )
-        if [ "${USE_CLO_LATEST_IMAGE:-false}" = true -a -n "${cloimg:-}" ] ; then
-            oc patch deploy/cluster-logging-operator --type=json \
-                --patch '[{"op":"replace","path":"/spec/template/spec/containers/0/image","value":"'"$cloimg"'"}]'
-        fi
-        if [ "${USE_EO_LATEST_IMAGE:-false}" = true -a -n "${eoimg:-}" ] ; then
-            oc -n $ESO_NS patch deploy/elasticsearch-operator --type=json \
-                --patch '[{"op":"replace","path":"/spec/template/spec/containers/0/image","value":"'"$eoimg"'"}]'
-            cureogen=$( oc -n $ESO_NS get deploy/elasticsearch-operator -o jsonpath='{.metadata.generation}' )
-            # doing the oc patch will restart eo - check to make sure it was restarted
-            eo_is_restarted() {
-                # wait until the old eo pod is not running and a new one is
-                if [ $starteogen -lt $cureogen ] && oc -n $ESO_NS get pods $eopod > /dev/null 2>&1 ; then
-                    return 1 # supposed to be restarted but old pod is still running
-                fi
-                oc -n $ESO_NS get pods | grep -q '^elasticsearch-operator-.* Running'
-            }
-            if ! wait_for_condition eo_is_restarted $DEFAULT_TIMEOUT > ${ARTIFACT_DIR}/test_output 2>&1 ; then
-                echo ERROR: elasticsearch-operator pod was not restarted
-                logging_err_exit
+    cloimg=$( construct_image_name cluster-logging-operator latest )
+    oc get deploy/cluster-logging-operator -o yaml | \
+    update_images_in_clo_yaml - $cloimg | \
+    oc replace --force -f -
+    eoimg=$( construct_image_name elasticsearch-operator latest )
+    if [ "${USE_EO_LATEST_IMAGE:-false}" = true -a -n "${eoimg:-}" ] ; then
+        oc -n $ESO_NS patch deploy/elasticsearch-operator --type=json \
+            --patch '[{"op":"replace","path":"/spec/template/spec/containers/0/image","value":"'"$eoimg"'"}]'
+        cureogen=$( oc -n $ESO_NS get deploy/elasticsearch-operator -o jsonpath='{.metadata.generation}' )
```

Comment:
is this field reliable?  My only concern is that since we constiually update the operator status this may change frequently, but maybe we update the status on the pod so this is ok.  @ewolinetz ?